frame/nomination-pools/src/lib.rs

```diff
@@ -0,0 +1,1525 @@
+				nominator,
+				state_toggler,
+				state,
+				account: pool_account.clone(),
+			},
+		)
+	}
+
+	/// Consume and put [`Self`] into storage.
+	fn put(self) {
+		let Self { account, points, depositor, root, nominator, state_toggler, state } = self;
+		BondedPools::<T>::insert(
+			account,
+			BondedPoolStorage { points, depositor, root, nominator, state_toggler, state },
+		);
+	}
+
+	/// Consume and remove [`Self`] from storage.
+	fn remove(self) {
+		BondedPools::<T>::remove(self.account);
+	}
+
+	/// Get the amount of points to issue for some new funds that will be bonded in the pool.
+	fn points_to_issue(&self, new_funds: BalanceOf<T>) -> BalanceOf<T> {
+		let bonded_balance =
+			T::StakingInterface::bonded_balance(&self.account).unwrap_or(Zero::zero());
+		points_to_issue::<T>(bonded_balance, self.points, new_funds)
+	}
+
+	/// Get the amount of balance to unbond from the pool based on a delegator's points of the pool.
+	fn balance_to_unbond(&self, delegator_points: BalanceOf<T>) -> BalanceOf<T> {
+		let bonded_balance =
+			T::StakingInterface::bonded_balance(&self.account).unwrap_or(Zero::zero());
+		balance_to_unbond::<T>(bonded_balance, self.points, delegator_points)
+	}
+
+	/// Issue points to [`Self`] for `new_funds`.
+	fn issue(&mut self, new_funds: BalanceOf<T>) -> BalanceOf<T> {
+		let points_to_issue = self.points_to_issue(new_funds);
+		self.points = self.points.saturating_add(points_to_issue);
+
+		points_to_issue
+	}
+
+	/// Check that the pool can accept a member with `new_funds`.
+	fn ok_to_join_with(&self, new_funds: BalanceOf<T>) -> Result<(), DispatchError> {
+		ensure!(self.state == PoolState::Open, Error::<T>::NotOpen);
+		let bonded_balance =
+			T::StakingInterface::bonded_balance(&self.account).unwrap_or(Zero::zero());
+		ensure!(!bonded_balance.is_zero(), Error::<T>::OverflowRisk);
+
+		let points_to_balance_ratio_floor = self
+			.points
+			// We checked for zero above
+			.div(bonded_balance);
+
+		// TODO make sure these checks make sense. Taken from staking design chat with Al
+
+		// Pool points can inflate relative to balance, but only if the pool is slashed.
+		//
+		// If we cap the ratio of points:balance so one cannot join a pool that has been slashed
+		// 90%,
+		ensure!(points_to_balance_ratio_floor < 10u32.into(), Error::<T>::OverflowRisk);
+		// while restricting the balance to 1/10th of max total issuance,
+		ensure!(
+			new_funds.saturating_add(bonded_balance) <
+				BalanceOf::<T>::max_value().div(10u32.into()),
+			Error::<T>::OverflowRisk
+		);
+		// then we can be decently confident the bonding pool points will not overflow
+		// `BalanceOf<T>`.
+		Ok(())
+	}
+
+	fn can_nominate(&self, who: &T::AccountId) -> bool {
+		*who == self.root || *who == self.nominator
+	}
+
+	fn can_kick(&self, who: &T::AccountId) -> bool {
+		(*who == self.root || *who == self.state_toggler) && self.state == PoolState::Blocked
+	}
+
+	fn is_destroying(&self) -> bool {
+		self.state == PoolState::Destroying
```

Comment:
Matches! Is a bit more idiomatic

Reply:
For some reason I am getting an error that it can't parse the `.` in `self.state`